File: src/main/java/com/sfh/pokeRogueBot/model/run/ChooseModifierDecision.java
package com.sfh.pokeRogueBot.model.run;

import com.sfh.pokeRogueBot.model.modifier.MoveToModifierResult;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class ChooseModifierDecision {

    private MoveToModifierResult freeItemToPick;
    private List<MoveToModifierResult> itemsToBuy;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/decisions/ChooseModifierDecision.kt
package com.sfh.pokeRogueBot.model.decisions

import com.sfh.pokeRogueBot.model.modifier.MoveToModifierResult

data class ChooseModifierDecision(
    val freeItemToPick: MoveToModifierResult?, //sometimes, no free item is picked and the free items are skipped
    val itemsToBuy: MutableList<MoveToModifierResult>
)